src/lib/EventTarget.ts

```diff
@@ -42,34 +42,33 @@ export default class EventTarget {
       this.pools.delete(poolName)
     }
 
-    this.removeTargetHandler(eventType)
-
-    if (this.pools.size > 0) {
-      this.addTargetHandler(eventType)
+    if (this.pools.size === 0) {
+      this.removeTargetHandler(eventType)
     }
   }
 
-  private createEmitter = (
-    eventType: string,
-    eventPools: Map<String, EventPool>,
-  ): EventListener => {
+  private createEmitter = (eventType: string): EventListener => {
     return (event: Event) => {
-      eventPools.forEach(pool => {
+      this.pools.forEach(pool => {
         pool.dispatchEvent(eventType, event)
       })
     }
   }
 
   private addTargetHandler(eventType: string) {
-    const handler = this.createEmitter(eventType, this.pools)
+    const handler = this.createEmitter(eventType)
 
     this.handlers.set(eventType, handler)
-    this.target.addEventListener(eventType, handler)
+    this.target.addEventListener(eventType, handler, true)
```

Comment:
https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener
`useCapture=true` the one of important changes there